app/lib/actions/pulp3/orchestration/repository/upload_content.rb

```diff
@@ -7,17 +8,25 @@ class UploadContent < Pulp3::Abstract
           def plan(repository, smart_proxy, file, unit_type_id)
             sequence do
               content_backend_service = SmartProxy.pulp_primary.content_service(unit_type_id)
-              duplicate_sha_path_content_list = content_backend_service.content_api.list(
+              duplicates_allowed = ::Katello::RepositoryTypeManager.find_content_type(unit_type_id).try(:duplicates_allowed)
+              if repository.generic? && duplicates_allowed
```

Comment:
maybe move this big if statement to a new function, find_duplciate_unit or something like that?